dali/operators/generic/one_hot.cc

```diff
@@ -74,13 +78,35 @@ TensorShape<> determine_shape(TensorShape<> in_shape, int num_classes, int axis,
   axis = axis < 0 ? in_shape.sample_dim() : axis;
   int outer_axes = axis;
   int inner_axes = in_shape.sample_dim() - axis;
-  auto outer =  in_shape.first(outer_axes);
+  auto outer = in_shape.first(outer_axes);
   auto inner = in_shape.last(inner_axes);
   return shape_cat(shape_cat(outer, num_classes), inner);
 }
 
 }  // namespace
 
+TensorLayout OneHot::GetOutputLayout(const HostWorkspace &ws, int placement_axis,
+                                     int output_sample_dim) const {
+  const auto &input = ws.template InputRef<CPUBackend>(0);
+  auto in_layout = input.GetLayout();
+  // .size method returns uint_8 which doesn't work well when 0 is printed in error message
+  int in_layout_size = in_layout.size();
+  if (!new_axis_name_) {
+    return {};
```

Comment:
You don't have to look at the inputs or extract layout to come to this conclusion. This can safely be the first thing this function does.

Reply:
Good point, fixed